passkey-authenticator/src/user_validation.rs

```diff
@@ -14,6 +29,19 @@ pub trait UserValidationMethod {
     /// `check_user_verification`. This will capture the user's consent to the operation.
     async fn check_user_presence(&self) -> bool;
 
+    /// Check for the user's presence and obtain consent for the operation. The operation may
+    /// also require the user to verify their identity.
+    ///
+    /// * `crdential` - Can be used to display additional information about the operation to the user.
+    /// * `presence` - Indicates whether the user's presence is required.
+    /// * `verification` - Indicates whether the user's identitfy should be verified.
+    async fn check_user(
+        &self,
+        credential: Option<Self::PasskeyItem>,
+        presence: bool,
+        verification: bool,
+    ) -> Result<UserCheck, Ctap2Error>;
```

Comment:
```suggestion
    ) -> Result<UserCheck, Ctap2Error> {
        Ok(UserCheck {
            presence,
            verification,
        })
    }
```

I feel like it might make sense for this to have a default implementation, to make it optional for current consumers of the API to implement. At the same time I tried that and it required adding more Send+Sync bounds, so maybe not worth it lol:
```
pub trait UserValidationMethod: Sync {
    type PasskeyItem: TryInto<Passkey> + Send;
...
```

Reply:
I considered this too, didn't know how to fix the errors, `UserValidationMethod: Sync` now I know! Thanks!